src/lib/spinel/CMakeLists.txt

```diff
@@ -72,7 +71,11 @@ target_include_directories(openthread-spinel-ncp PUBLIC ${OT_PUBLIC_INCLUDES} PR
 target_include_directories(openthread-spinel-rcp PUBLIC ${OT_PUBLIC_INCLUDES} PRIVATE ${COMMON_INCLUDES})
 
 target_sources(openthread-spinel-ncp PRIVATE ${COMMON_SOURCES})
-target_sources(openthread-spinel-rcp PRIVATE ${COMMON_SOURCES})
+target_sources(openthread-spinel-rcp
+    PRIVATE
+        ${COMMON_SOURCES}
+        radio_spinel.cpp
```

Comment:
@zhanglongxia - why is this file need for RCP builds?  I believe all the functions in this file are for the host.

Reply:
I agree. I don't think we should include here.
`openthread-spinel-rcp` seem to be used when building RCP firmware and `radio_spinel` module is for host side which is interacting with RCP.

Reply:
The host also uses the lib [openthread-spinel-rcp](https://github.com/openthread/openthread/blob/main/src/posix/platform/CMakeLists.txt#L157). So I put `radio_spinel.cpp` to `openthread-spinel-rcp`. 

You are right, the file `radio_spinel.cpp` is not needed by building the RCP. How about adding a new lib `openthread-radio-spinel` to include the file `radio_spinel.cpp`?

Reply:
> So we can consider adding it directly in `posix` build itself for now?

This makes sense to me.

Reply:
> So I think it is better to add a new lib `openthread-radio-spinel` to include the file `radio_spinel.cpp`.

SGTM.

Reply:
Updated